### ISSUE TITLE
HTML Form Widget crashes QGIS

### ISSUE THREAD

**Issue by PedroVenancio** [qgis/QGIS#33390]
Using a simple HTML Form Widget in a layer with a joined table, leads to a QGIS crash, when the join is configured with the **Dynamic form** option active.

To reproduce:

1. Open this sample project in QGIS 3.10.1:
[HTML_Widget.zip](https://github.com/qgis/QGIS/files/3961540/HTML_Widget.zip)

2. Use the Identify Features tool and check that everything is ok.

3. Open 'concelhos' Properties -> Joins -> Edit selected join -> check Dynamic form option
![dynamic_form](https://user-images.githubusercontent.com/1331581/70814059-9cd09e00-1dc2-11ea-9cc9-992cecd1d795.jpg)

4. Use the Identify tool again -> QGIS crashes.

```
Stack Trace
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QMetaObject::activate :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QMetaObject::activate :
QLineEdit::qt_static_metacall :
QMetaObject::activate :
QWidgetLineControl::finishChange :
QWidgetLineControl::internalSetText :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgsMapLayerComboBox::qt_static_metacall :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QMetaObject::activate :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgisApp::actionOpenFieldCalculator :
QgsMapLayerComboBox::qt_static_metacall :
QWidget::event :
QFrame::event :
QGraphicsView::viewportEvent :
QCoreApplicationPrivate::sendThroughObjectEventFilters :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsCircularString::insertVertex :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsCircularString::insertVertex :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
QgisApp::actionOpenFieldCalculator :
BaseThreadInitThunk :
RtlUserThreadStart :

QGIS Info
QGIS Version: 3.10.1-A Coruna
QGIS code revision: e8f377f62d
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.1.0dev
Running against GDAL: 3.1.0dev

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.18363
```

**Comment by PedroVenancio** [comment]
Thank you very much @elpaso !

**Comment by elpaso** [comment]
@PedroVenancio you're welcome, can you please confirm that the crash is gone and the behavior of the identify operation is what is expected?

**Comment by PedroVenancio** [comment]
@elpaso I confirm the fix in master now! I'll also test the next 3.10.1 nightly build.

Thank you very much!
